prov/efa/test/efa_unit_test_mocks.c

```diff
@@ -155,3 +158,49 @@ int __wrap_efadv_query_device(struct ibv_context *ibv_ctx, struct efadv_device_a
 {
 	return g_efa_unit_test_mocks.efadv_query_device(ibv_ctx, attr, inlen);
 }
+
+#if HAVE_EFADV_CQ_EX
+uint32_t efa_mock_ibv_read_src_qp_return_mock(struct ibv_cq_ex *current)
+{
+	return mock();
+}
+
+uint32_t efa_mock_ibv_read_byte_len_return_mock(struct ibv_cq_ex *current)
+{
+	return mock();
+};
+
+uint32_t efa_mock_ibv_read_slid_return_mock(struct ibv_cq_ex *current)
+{
+	return mock();
+}
+
+int efa_mock_efadv_wc_read_unknown_ah(struct efadv_cq *efadv_cq, union ibv_gid *sgid)
+{
+	memcpy(sgid->raw, (uint8_t *)mock(), sizeof(sgid->raw));
+	/* Must return a negative value for unknown AH */
+	return -ENOENT;
+};
+
+struct efadv_cq *__real_efadv_cq_from_ibv_cq_ex(struct ibv_cq_ex *ibvcqx);
+
+struct efadv_cq *__wrap_efadv_cq_from_ibv_cq_ex(struct ibv_cq_ex *ibvcqx)
+{
+	return g_efa_unit_test_mocks.efadv_cq_from_ibv_cq_ex(ibvcqx);
+}
+
+struct efadv_cq *efadv_cq_from_ibv_cq_ex_return_mock(struct ibv_cq_ex *ibvcqx)
```

Comment:
I am not sure why this mock is needed?

Reply:
We need to mock the returned `struct efadv_cq` value, and then mock `efadv_cq.wc_read_qh(...)`

Reply:
I see. 

Can we call the real `efadv_cq_from_ibv_cq_ex` to get an `efqdv_cq`, and set the result's `wc_read_ah` to what we want?

Reply:
That's a good idea. Then we don't need to mock this function.